File: Common/src/main/java/com/common/dto/response/WrapperApiResponse.java
package com.common.dto.response;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.time.OffsetDateTime;

/**
 * Lớp wrapper dùng để đóng gói 1 response trả về client
 * Created at 06/06/2026
 *
 * @param status
 * @param message
 * @param data
 * @param timestamp
 * @author txhoan
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record WrapperApiResponse(
        int status,
        String message,
        Object data,
        OffsetDateTime timestamp
) {
}

File: be/.Net/Swe.Common/Model/WrapperApiResponse.cs
using System;

namespace Swe.Common.Model;

/// <summary>
/// Lớp wrapper dùng để đóng gói 1 response trả về client.
/// </summary>
/// <created_at>2026-08-03</created_at>
/// <author>txhoan</author>
public class WrapperApiResponse
{
    public int Status { get; set; }
    public string Message { get; set; } = string.Empty;
    public object? Data { get; set; }
    public DateTime Timestamp { get; set; } = DateTime.UtcNow;

    public WrapperApiResponse() { }

    public WrapperApiResponse(int status, string message, object? data)
    {
        Status = status;
        Message = message;
        Data = data;
        Timestamp = DateTime.UtcNow;
    }
}
